__tests__/diff-snapshot.spec.js

```diff
@@ -219,12 +223,38 @@ describe('diff-snapshot', () => {
       expect(mockWriteFileSync).toHaveBeenCalledTimes(1);
     });
 
+    it('should write a received image if the test fails and storeReceivedOnFailure = true', () => {
+      const diffImageToSnapshot = setupTest({ snapshotExists: true, pixelmatchResult: 5000 });
+      const result = diffImageToSnapshot({
+        receivedImageBuffer: mockFailImageBuffer,
+        snapshotIdentifier: mockSnapshotIdentifier,
+        snapshotsDir: mockSnapshotsDir,
+        storeReceivedOnFailure: true,
+        receivedDir: mockReceivedDir,
```

Comment:
Can we also add a case for when `receivedDir` is undefined

Reply:
The check for undefined `receivedDir` and other custom paths is handled in the `index.js` file. There did not seem to be a specific test that checked for default values, so I added that.